core/src/test/java/org/infinispan/xsite/irac/IracAlwaysRemoveConflictTest.java

```diff
@@ -0,0 +1,151 @@
+
+   protected IracAlwaysRemoveConflictTest() {
+      this.iracManagerList = new ArrayList<>(N_SITES * CLUSTER_SIZE);
+   }
+
+   public void testPutIfAbsent(Method method) {
+      doTest(method, TestOperation.PUT_IF_ABSENT);
+   }
+
+   public void testPut(Method method) {
+      doTest(method, TestOperation.PUT);
+   }
+
+   public void testReplace(Method method) {
+      doTest(method, TestOperation.REPLACE);
+   }
+
+   public void testConditionalReplace(Method method) {
+      doTest(method, TestOperation.REPLACE_CONDITIONAL);
+   }
+
+   public void testRemove(Method method) {
+      doTest(method, TestOperation.REMOVE);
+   }
+
+   public void testConditionalRemove(Method method) {
+      doTest(method, TestOperation.REMOVE_CONDITIONAL);
+   }
+
+   @Override
+   protected int defaultNumberOfSites() {
+      return N_SITES;
+   }
+
+   @Override
+   protected int defaultNumberOfNodes() {
+      return CLUSTER_SIZE;
+   }
+
+   @Override
+   protected ConfigurationBuilder defaultConfigurationForSite(int siteIndex) {
+      ConfigurationBuilder builder = getDefaultClusteredCacheConfig(CacheMode.DIST_SYNC);
+      builder.sites().mergePolicy(XSiteMergePolicy.ALWAYS_REMOVE);
+      for (int i = 0; i < N_SITES; ++i) {
+         if (i == siteIndex) {
+            //don't add our site as backup.
+            continue;
+         }
+         builder.sites()
+               .addBackup()
+               .site(siteName(i))
+               .strategy(BackupConfiguration.BackupStrategy.ASYNC);
+      }
+      return builder;
+   }
+
+   @Override
+   protected GlobalConfigurationBuilder defaultGlobalConfigurationForSite(int siteIndex) {
+      GlobalConfigurationBuilder builder = GlobalConfigurationBuilder.defaultClusteredBuilder();
+      builder.serialization().addContextInitializer(TestDataSCI.INSTANCE);
+      return builder;
+   }
+
+   @AfterMethod(alwaysRun = true)
+   @Override
+   protected void clearContent() throws Throwable {
+      iracManagerList.forEach(iracManager -> iracManager.disable(ManualIracManager.DisableMode.DROP));
+      super.clearContent();
+   }
+
+   @Override
+   protected void afterSitesCreated() {
+      for (int i = 0; i < N_SITES; ++i) {
+         for (Cache<?, ?> cache : caches(siteName(i))) {
+            iracManagerList.add(ManualIracManager.wrapCache(cache));
+         }
+      }
+   }
+
+   private void doTest(Method method, TestOperation testConfig) {
+      final String key = TestingUtil.k(method, 0);
+      final String initialValue = testConfig.requiresPreviousValue() ? TestingUtil.v(method, 0) : null;
+
+      //init cache if needed!
+      if (testConfig.requiresPreviousValue()) {
+         cache(siteName(0), 0).put(key, initialValue);
+      }
+      eventuallyAssertInAllSitesAndCaches(cache -> Objects.equals(initialValue, cache.get(key)));
+
+      //disable xsite so each site won't send anything to the others
+      iracManagerList.forEach(ManualIracManager::enable);
```

Comment:
I didn't notice this when you first added `ManualIracManager`, but the comment would be much easier to understand (or maybe not even needed) if you called the method `disableXSite()` instead of `enable()`.